Clean up intro; add quickstart guide (#81)
teamkeel/docs: pages/quickstart.mdx

keel init
```

This will create a few files for us, specifically:

- `schema.keel` - The file that represents our application's backend.
- `keelconfig.yaml` - Configuration for our backend around secrets, authentication, and more. We'll cover this in more detail later.

Great, we've got a Keel schema. Let's populate it.

## The Keel Schema

The Keel schema is a custom language that describes your application's backend. It's a declarative language that allows you to describe your application's data models, actions your application can do on those models, authentication, and more.

Let's explore this by creating a simple application that allows users to create and view posts.

<Callout type="info" emoji="ℹ️">
  We've created an AI assistant called [KeelGPT](https://chat.openai.com/g/g-0xwNB1Azt-keelgpt) to help you navigate the Keel schema. It can generate schemas for you, answer questions, and more. It may be helpful as you follow the schema below.
</Callout>

Paste this into your `schema.keel` file:

```keel
// We define a model
model User {

    // A model has fields
    fields {
        identity Identity @unique
        name Text
        email Text
        posts Post[]
    }

>     // A model also has actions you can do to it, like

Reviewer: ```suggestion
    // A model also has actions which can be used via API's
```

Reviewer 2: I don't think it's blocking but if you're in there doing the other changes can you update as I think it works better.

Author: Are we sure about `API's` or do we mean `APIs`?

Reviewer 2: Oh yeah defo `APIs` not `API's`

    actions {

        // Create
        create createUser() with (name, email) {
            @set(user.identity = ctx.identity)
        }

>         // Get a single unit

Reviewer: ```suggestion
        // Get a single user
```

# We're calling the authenticate mutation
  authenticate(
    # It takes an `input` argument
    input: {
        # We're authenticating with an email and password
        createIfNotExists: true, 
        emailPassword: {
            email: "test@keel.xyz",
            password: "superSecr3t!"
        }
    }
  ) {
    # We get back 
    token # an authentication token
    identityCreated # whether or not the identity was created
  }
}
```

If all went well, we should receive a response that looks something like this:

```json
{
  "data": {
    "authenticate": {
      "identityCreated": true,
      "token": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrZWVsIiwic3ViIjoiMll0ckZPdnFjQmQyZzFKRFQzOE9WQW9qWmMxIiwiZXhwIjoxNzAxNDQyMDEyLCJpYXQiOjE3MDEzNTU2MTJ9.bFBsXXJIaVy4ELRKInBKKQoFEVtAwJ1DUNH9oCwo0aT-5r5lQA6kEXJUrDK35Ini-0rKJfahwz-FcOQ6uLToIxuvoWeTtCLvVjEeBVcJWvpEASxXfntnIvn3oZBOhR8icExbPecbKE0dMmxRxvWmrjr3XDD9gUeWpj1MO4ihCgCszHRFJRjcMdy-3QzQ81LZXn0_SeXVpMgdfKIx7HLR95B2mnQi45isNreg9_lpzpXNIzH6r6YAmgamEmZIBts7e908VI2GeMEgB_ebgr2nN5s9R7X1rOSuXs31dB7LBsJgf2xYJ8KkF8O4SUV3rVZl1smJeBUsuMS6aWvy6cnwkQ"
    }
  }
}
```

That `token` is what will let us use the rest of the API.

> Great! Now we have an **API user**. Let's use it to create an **application user**. To do this, we'll run the following query:

Reviewer: ```suggestion
Great! Now we have an **identity**. Let's use it to create a `User`. To do this, we'll run the following query:
```

Author: It'd be a little bit problematic to use the term Identity without clarifying what it is IMO. Unless we do something like

```suggestion
Great! Now we have an **Identity**. An Identity represents a user at the API level. Let's use it to create `User` from our data model at the application level. To do this, we'll run the following query:
```

Reviewer 2: Let's go with the "identity" suggestion and then link that out to the identity page rather than trying to explain it inline.
